CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+
+## Preparation
+
+If you are brand new to software developments or robotics, you might find our training materials useful. We use these training materials to onboard approximately 20 freshman every year. (contributions are welcome to this repository as well!)
+
+[RoboJackets/robocup-training](https://github.com/RoboJackets/robocup-training)
+
+This repo contains most of our slides, starter documentation, and sample issues. If you find anything confusing (or something that can be improved) please open an issue on that repository.
+
+For people who already have their bearings, [we have autogenerated documentation sites for most robocup projects](https://robojackets.github.io/robocup-software/).
+
+## Getting Started
+
+The first step to contributing is to find an issue. We reccoment searching our issue list for [intro](https://github.com/RoboJackets/robocup-software/issues?q=is%3Aopen+is%3Aissue+label%3A%22exp+%2F+intro%22) or [beginner](https://github.com/RoboJackets/robocup-software/issues?q=is%3Aopen+is%3Aissue+label%3A%22exp+%2F+beginner%22) issues first. These links only link to these repos, however, we maintain many different repos that you might want to check out to try to find something to work on:
+- [RoboJackets/robocup-software](https://github.com/RoboJackets/robocup-software)
+- [RoboJackets/robocup-firmware](https://github.com/RoboJackets/robocup-firmware)
+- [RoboJackets/robocup-common](https://github.com/RoboJackets/robocup-common)
+- [RoboJackets/rrt](https://github.com/RoboJackets/rrt)
+- [RoboJackets/robocup-pcb](https://github.com/RoboJackets/robocup-pcb)
+- [RoboJackets/grSim](https://github.com/RoboJackets/grSim) (forked from [grSim](https://github.com/mani-monaj/grSim))
+
+For simplicity, this guide will cover only RoboJackets/robocup-software directly (with links), but the tips found here can help you contribute to any of these projects.
+
+Once you have an issue, and a basic idea of it's solution, its time to get the current version of the software running.
+
+## Machine Setup
+
+RoboJackets-RoboCup projects run best on Ubuntu, but can run on any linux distribution if you set up the dependencies yourself.
+
+First clone the repo you want to work on. Eg: `git clone https://github.com/RoboJackets/robocup-software`.
+
+For Ubuntu, we provide an [automated setup script](https://github.com/RoboJackets/robocup-software/blob/master/util/ubuntu-setup). To install it, you can follow the instructions in [this section of our README](https://github.com/RoboJackets/robocup-software#setup).
+
+Once the setup is completed, you should be able to run `make` to compile our project. For robocup-software, you can then run `make run-sim` to run our simulator and soccer, and play around with the running program.
+
+### Note about Submodules
```

Comment:
It might be good to mention that you can setup git hooks to do this automatically. Submodules always give git-newcomers trouble, but with the hooks installed, they're pretty transparent.

https://github.com/RoboJackets/robocup-software/tree/master/util/git-hooks